core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DedupAggregator.java

```diff
@@ -201,11 +202,13 @@ private void validateSchema(Schema inputSchema, List<String> uniqueFields,
                         null)
                 .withConfigProperty("filterOperation");
       }
-      Schema.Type fieldType = field.getSchema().isNullable() ?
-              field.getSchema().getNonNullable().getType() : field.getSchema().getType();
+      Schema fieldSchema = field.getSchema().isNullable() ? field.getSchema().getNonNullable() : field.getSchema();
+      Schema.Type fieldType = fieldSchema.getType();
+      Schema.LogicalType logicalFieldType = fieldSchema.getLogicalType();
       if ((function.getFunction() == DedupConfig.Function.MAX || function.getFunction() == DedupConfig.Function.MIN)
-        && !ALLOWED_SCHEMA_TYPES.contains(fieldType)) {
-        collector.addFailure(String.format("Unsupported filter operation %s(%s): Field '%s' is non numeric ",
+        && (!ALLOWED_SCHEMA_TYPES.contains(fieldType) && logicalFieldType == null)) {
```

Comment:
We should probably restrict logical types to the types that we want to support, checking that `ALLOWED_SCHEMA_TYPES` or `ALLOWED_LOGICAL_TYPES` is what we want it to be.